In this question, you will be analyzing a subset of the imdb movies database which consists of the top 10k movies. Download the file called `imdb_top_10000.txt` in the homework folder on smcmba  and follow the next steps to generate a good analysis report based on the given file.<br>
First step is load the file using pandas. The columns represent `['imdbID', 'title', 'year', 'score', 'votes', 'runtime', 'genres']`. You need to decide on the correct separator.
Display the top and bottom 7 elements of the table

In [36]:
#solve here
import pandas as pd
import numpy as np

df=pd.read_csv('imdb_top_10000.txt', names=['imdbID', 'title', 'year', 'score', 'votes', 
                                            'runtime', 'genres'],sep="\t" )
df.head(6),df.tail(6)

(      imdbID                            title  year  score   votes    runtime  \
 0  tt0111161  The Shawshank Redemption (1994)  1994    9.2  619479  142 mins.   
 1  tt0110912              Pulp Fiction (1994)  1994    9.0  490065  154 mins.   
 2  tt0137523                Fight Club (1999)  1999    8.8  458173  139 mins.   
 3  tt0133093                The Matrix (1999)  1999    8.7  448114  136 mins.   
 4  tt1375666                 Inception (2010)  2010    8.9  385149  148 mins.   
 5  tt0109830              Forrest Gump (1994)  1994    8.7  368994  142 mins.   
 
                              genres  
 0                       Crime|Drama  
 1                    Crime|Thriller  
 2            Drama|Mystery|Thriller  
 3           Action|Adventure|Sci-Fi  
 4  Action|Adventure|Sci-Fi|Thriller  
 5              Comedy|Drama|Romance  ,
          imdbID                          title  year  score  votes    runtime  \
 9994  tt0118635                  Aprile (1998)  1998    6.7   1358 

Is there any Nans in the data? What are the number of nulls in the data? Clean up the data and remove the rows that contains the nulls
<br> **Bonus if you display only the rows that has Nans**

In [37]:
df.info()#only one row has nulls

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 7 columns):
imdbID     10000 non-null object
title      10000 non-null object
year       10000 non-null int64
score      10000 non-null float64
votes      10000 non-null int64
runtime    10000 non-null object
genres     9999 non-null object
dtypes: float64(1), int64(2), object(4)
memory usage: 547.0+ KB


In [38]:
df[pd.isnull(df).any(axis =1)]#show row with. nulls 

imdbID             title  year  score  votes   runtime genres
7917  tt0990404  Chop Shop (2007)  2007    7.2   2104  84 mins.    NaN

In [39]:
# clean data
new_df = df.dropna()
new_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9999 entries, 0 to 9999
Data columns (total 7 columns):
imdbID     9999 non-null object
title      9999 non-null object
year       9999 non-null int64
score      9999 non-null float64
votes      9999 non-null int64
runtime    9999 non-null object
genres     9999 non-null object
dtypes: float64(1), int64(2), object(4)
memory usage: 624.9+ KB


The column called runtime contains the duration of the movie with the word mins next to it (such as 142 mins). We need to have the duration as just numbers without the word mins so we can do some statistical analysis on the movies durations. Use the pandas function `str.split` to split the runtime column into two (number and the word mins). Then add the one that has the numbers to the data as a new column called duration. For example, the rows that had runtime 142 mins, will have a new column called duration which contains 142

In [40]:
# create column duration from runtime
new_df['duration'] = new_df['runtime'].apply(lambda x : x.split()[0])

/Users/claire_kim/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


In [41]:
new_df.head()

imdbID                            title  year  score   votes    runtime  \
0  tt0111161  The Shawshank Redemption (1994)  1994    9.2  619479  142 mins.   
1  tt0110912              Pulp Fiction (1994)  1994    9.0  490065  154 mins.   
2  tt0137523                Fight Club (1999)  1999    8.8  458173  139 mins.   
3  tt0133093                The Matrix (1999)  1999    8.7  448114  136 mins.   
4  tt1375666                 Inception (2010)  2010    8.9  385149  148 mins.   

                             genres duration  
0                       Crime|Drama      142  
1                    Crime|Thriller      154  
2            Drama|Mystery|Thriller      139  
3           Action|Adventure|Sci-Fi      136  
4  Action|Adventure|Sci-Fi|Thriller      148

What is min, max, average and variance of the durations and which movies have the longest durations?

In [42]:
# change duration datatype
new_df.duration = new_df.duration.astype(int, errors = 'ignore')
#show information about dataset to see if duration is int
new_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9999 entries, 0 to 9999
Data columns (total 8 columns):
imdbID      9999 non-null object
title       9999 non-null object
year        9999 non-null int64
score       9999 non-null float64
votes       9999 non-null int64
runtime     9999 non-null object
genres      9999 non-null object
duration    9999 non-null int64
dtypes: float64(1), int64(3), object(4)
memory usage: 703.1+ KB


In [48]:
new_df.duration.describe() #min,max and average

count    9999.000000
mean      103.580358
std        26.629310
min         0.000000
25%        93.000000
50%       102.000000
75%       115.000000
max       450.000000
Name: duration, dtype: float64

In [49]:
new_df.duration.var() #variance

709.1201422602626

What are the years provided in the given file?

In [59]:
# Solve here
# hint use unique and sort the outcome
years = sorted(new_df.year.unique())
years[0:10] #showing first 10 yeears

[1950, 1951, 1952, 1953, 1954, 1955, 1956, 1957, 1958, 1959]

Is there any duplicates in the data?

In [61]:
# solve here
new_df[new_df.duplicated()]

Empty DataFrame
Columns: [imdbID, title, year, score, votes, runtime, genres, duration]
Index: []

In [65]:
n = new_df.duplicated()
n.unique()#no duplicates

array([False])

Calculate the number of movies produced each year then plot number of movies vs years, what is the year that has the highest movies?
Calculate this with the code and verify it with the graph

In [45]:
# Solve here
